walk/cache/cache.go

```diff
@@ -14,26 +15,33 @@ const (
 	bucketPaths = "paths"
 )
 
-func Open(root string) (*bolt.DB, error) {
-	var (
-		err  error
-		path string
-	)
-
-	// Otherwise, the database will be located in `XDG_CACHE_DIR/treefmt/eval-cache/<name>.db`, where <name> is
-	// determined by hashing the treeRoot path.
-	// This associates a given treeRoot with a given instance of the cache.
+// Path returns a unique local cache file path for the given root string, using its SHA-256 hash.
+func Path(root string) (string, error) {
 	digest := sha256.Sum256([]byte(root))
 
 	name := hex.EncodeToString(digest[:])
-	if path, err = xdg.CacheFile(fmt.Sprintf("treefmt/eval-cache/%v.db", name)); err != nil {
-		return nil, fmt.Errorf("could not resolve local path for the cache: %w", err)
+
+	path, err := xdg.CacheFile(fmt.Sprintf("treefmt/eval-cache/%v.db", name))
+	if err != nil {
+		return "", fmt.Errorf("could not resolve local path for the cache: %w", err)
+	}
+
+	return path, nil
+}
+
+// Open initialises and opens a Bolt database for the specified root path.
+// Returns a pointer to the opened database or an error if initialisation fails.
+func Open(root string) (*bolt.DB, error) {
+	// determine the db location
+	path, err := Path(root)
+	if err != nil {
+		return nil, err
 	}
 
 	// open db
 	db, err := bolt.Open(path, 0o600, &bolt.Options{Timeout: 1 * time.Second})
 	if err != nil {
-		return nil, fmt.Errorf("failed to open cache db: %w", err)
+		return nil, fmt.Errorf("failed to open cache db at %s: %w", path, err)
```

Comment:
(Not a blocker for this PR) Does bbolt have any mechanism for reporting information on *what* has the cache locked?

Reply:
Don't know, worth looking into.